### PR TITLE
Item 6118: Assay Design Create/Update

### PR DESCRIPTION
**Pull request by cnathe**

### CODE REVIEW

File: experiment/src/client/AssayDesigner/AssayDesigner.tsx [hunk 1]
```diff
@@ -31,41 +34,104 @@ export class App extends React.Component<any, State> {
...
-            });
+        // if URL has a protocol RowId, look up the assay design info
+        if (protocolId) {
+            fetchProtocol(protocolId)
+                .then((model) => {
+                    this.setState({
+                        model,
+                        isLoading: false
+                    });
+                })
```
Comment (labkey-martyp) [inline experiment/src/client/AssayDesigner/AssayDesigner.tsx]: I'm guilty of this myself (or of not totally understanding why we're doing it), but should this setState have a function as a parameter?

File: experiment/src/client/AssayDesigner/AssayDesigner.tsx [hunk 2]
```diff
@@ -31,41 +34,104 @@ export class App extends React.Component<any, State> {
...
+        if (this.state.dirty) {
+            event.returnValue = 'Changes you made may not be saved.';
+        }
+    };
+
+    navigate(defaultUrl: string) {
+        const { returnUrl } = this.state;
+        // this.setState(() => ({dirty: false}), () => {
+            window.location.href = returnUrl || defaultUrl;
+        // });
```
Comment (labkey-martyp) [inline experiment/src/client/AssayDesigner/AssayDesigner.tsx]: Can this commented out code be deleted?

File: experiment/src/client/AssayDesigner/AssayDesigner.tsx [hunk 3]
```diff
@@ -75,19 +141,12 @@ export class App extends React.Component<any, State> {
             return <LoadingSpinner/>
         }
 
+        if (readOnly) {
+            return <>{this.renderReadOnlyView()}</>
```
Comment (labkey-martyp) [inline experiment/src/client/AssayDesigner/AssayDesigner.tsx]: They may be necessary, but not sure what the <></> are for here.  Shouldn't renderReadOnlyView return a single tsx component?

File: experiment/src/client/AssayDesigner/AssayDesigner.tsx [hunk 4]
```diff
@@ -75,19 +141,12 @@ export class App extends React.Component<any, State> {
...
-                        <div className={"panel-title"}>{protocol.name}</div>
-                    </Panel.Heading>
-                    <Panel.Body>
-                        <p>Provider: {protocol.providerName}</p>
-                        <p>Description: {protocol.description}</p>
-                    </Panel.Body>
-                </Panel>
-                {this.renderDomains()}
-            </>
+            <>{this.renderDesignerView()}</>
```
Comment (labkey-martyp) [inline experiment/src/client/AssayDesigner/AssayDesigner.tsx]: Same as above

File: experiment/src/client/AssayDesigner/AssayDesigner.tsx (hunk 1 above)
Comment (cnathe) [reply experiment/src/client/AssayDesigner/AssayDesigner.tsx]: Yes, fixed.

File: experiment/src/client/AssayDesigner/AssayDesigner.tsx (hunk 2 above)
Comment (cnathe) [reply experiment/src/client/AssayDesigner/AssayDesigner.tsx]: Actually, I am going to finish the implementation for the dirty state and then uncomment this. Thanks for the reminder.

File: experiment/src/client/AssayDesigner/AssayDesigner.tsx (hunk 3 above)
Comment (cnathe) [reply experiment/src/client/AssayDesigner/AssayDesigner.tsx]: Yeah, this syntax is a little confusing to read. The `<></>` is necessary with how I have it, but I just consolidated this with the lines below into a more readable if/else.

**Pull request merged**